network/src/main/java/io/bisq/network/p2p/network/SynchronizedProtoOutputStream.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of Bisq.
+ *
+ * Bisq is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * Bisq is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with Bisq. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package io.bisq.network.p2p.network;
+
+import io.bisq.common.proto.network.NetworkEnvelope;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.concurrent.ThreadSafe;
+import java.io.OutputStream;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+@ThreadSafe
+class SynchronizedProtoOutputStream extends ProtoOutputStream {
+    private static final Logger log = LoggerFactory.getLogger(SynchronizedProtoOutputStream.class);
+
+    private final ExecutorService executorService;
+
+    SynchronizedProtoOutputStream(OutputStream delegate, Statistic statistic) {
+        super(delegate, statistic);
+        this.executorService = Executors.newSingleThreadExecutor();
+    }
+
+    @Override
+    void writeEnvelope(NetworkEnvelope envelope) {
+        Future<?> future = executorService.submit(() -> super.writeEnvelope(envelope));
+        try {
+            future.get();
+        } catch (InterruptedException e) {
+            Thread currentThread = Thread.currentThread();
+            currentThread.interrupt();
+            log.error("Thread " + currentThread + " was interrupted", e);
+            throw new BisqRuntimeException("Failed to write envelope", e);
```

Comment:
@ManfredKarrer not sure if an exception should be thrown in this case. An Interrupted exception may occur on shutdown of executor service - maybe just returning from `writeEnvelope` is better.